libs/graphslam/include/mrpt/graphslam/CGraphSlamEngine.h

```diff
@@ -735,6 +773,16 @@ class CGraphSlamEngine : public mrpt::utils::COutputLogger {
 		 */
 		double m_dataset_grab_time;
 
+		/**\brief First recorded timestamp in the dataset. */
+		mrpt::system::TTimeStamp* m_init_timestamp;
```

Comment:
It may be safe, but easy to break in the future if you or someone else updates the code: using pointers to TTimeStamp seems unnecesarily dangerous to me: I normally use a plain TTimeStamp initialized to INVALID_TIMESTAMP to reflect it's invalid. A pointer may dangle if the pointee get out of context, etc. It may become a nightmare :-)